src/components/Login/Login.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState, useEffect } from 'react';
+import { useLocation, Navigate } from 'react-router-dom';
+import { instanceOf } from 'prop-types';
+import {
+  Flex,
+  FormControl,
+  FormLabel,
+  Input,
+  Button,
+  Heading,
+  Link,
+  Stack,
+  Alert,
+  AlertIcon,
+  Box,
+  Spinner,
+} from '@chakra-ui/react';
+import styles from './Login.module.css';
+import { Cookies, withCookies, clearCookies } from '../../utils/CookieUtils';
+import { logInWithEmailAndPassword, useNavigate, refreshToken } from '../../utils/AuthUtils';
+import { PNPBackend } from '../../utils/utils';
+
+const userIsAuthenticated = async cookies => {
+  try {
+    const accessToken = await refreshToken(cookies);
+    if (!accessToken) {
+      return false;
+    }
+    const loggedIn = await PNPBackend.get(`/auth/verifyToken/${accessToken}`);
+    return loggedIn.status === 200;
+  } catch (err) {
+    clearCookies(cookies);
+    return false;
+  }
+};
+
+const Login = ({ cookies }) => {
+  const navigate = useNavigate();
+  const [isLoading, setIsLoading] = useState(true);
+  const [isAuthenticated, setIsAuthenticated] = useState(false);
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [errorMessage, setErrorMessage] = useState();
+
+  const { search } = useLocation();
+  const signup = new URLSearchParams(search).get('signup');
+
+  useEffect(() => {
+    const checkUserAuthentication = async () => {
+      const authenticated = await userIsAuthenticated(cookies);
+      setIsAuthenticated(authenticated);
+      setIsLoading(false);
+    };
+    checkUserAuthentication();
+  }, []);
+
+  /**
+   * This function handles logging in with email/password (standard log in)
+   * If the user signs in successfully, they are redirected to the dashboard, otherwise they are redirected to the login screen
+   * @param {Event} e
+   */
+  const handleSubmit = async e => {
+    try {
+      e.preventDefault();
+      await logInWithEmailAndPassword(email, password, '/', navigate, cookies);
+    } catch (err) {
+      setErrorMessage(err.message);
```

Comment:
Saying something about "FireBase error auth" doesn't make sense to the user, so see if you could parse the error returned to be more user-friendly.